crates/host/src/lib.rs

```diff
@@ -73,16 +74,29 @@ impl<'a> TryFrom<&'a str> for ResourceRef<'a> {
                         .to_file_path()
                         .map(Self::File)
                         .map_err(|()| anyhow!("failed to convert `{url}` to a file path")),
-                    "bindle" => s
-                        .strip_prefix("bindle://")
-                        .map(Self::Bindle)
-                        .context("invalid Bindle reference"),
-                    "oci" => s
-                        .strip_prefix("oci://")
-                        .map(Self::Oci)
-                        .context("invalid OCI reference"),
-                    // TODO: Support other schemes
-                    scheme => bail!("unsupported scheme `{scheme}`"),
+                    "bindle" => {
+                        // Note: bindle is not a scheme, but using this as a prefix takes out the guesswork
+                        s.strip_prefix("bindle://")
+                            .map(Self::Bindle)
+                            .context("invalid Bindle reference")
+                    }
+                    "oci" => {
+                        // Note: oci is not a scheme, but using this as a prefix takes out the guesswork
+                        s.strip_prefix("oci://")
+                            .map(Self::Oci)
+                            .context("invalid OCI reference")
+                    }
+                    scheme @ ("http" | "https") => {
+                        debug!(%url, "interpreting reference as OCI");
+                        s.strip_prefix(&format!("{scheme}://"))
+                            .map(Self::Oci)
+                            .context("invalid OCI reference")
+                    }
+                    _ => {
```

Comment:
Nit: This is a minor concern since I consider this overall an improvement: but won’t this cause error messages to the user that say things like “invalid OCI reference” when they didn’t think they passed an OCI reference?

Reply:
I'm not sure if you meant to comment a couple lines up? The `.context("invalid OCI reference")` lines should never be processed in practice, since they're handling the `None` case of `s.strip_prefix(...)`, and we've already matched against the `scheme` of the parsed string

The one potentially odd case for users is the fallback `_`, where we will print a debug message saying `unknown scheme in reference, assuming OCI` for URLs like `registry:5000/v2/echo:0.1.0`, but I think this is the best we can do without actually adding logic to intelligently query the server

Reply:
So I did mean to leave the comment here. I know that right now it will debug print the message, but most people won't have it set to debug. My main concern is that a "bare" reference (not passed with a schema) will be then passed to the oci-distribution crate, and if it is an invalid reference it will then say "invalid OCI reference" or "unknown image <reference>." Does that make any more sense?

Reply:
Synced through this with @connorsmith256 offline and like we thought, there isn't much we can do here for the weird error cases